### PR TITLE
2.5D stuff

### PR DESCRIPTION
**Pull request by oterral**
This PR adds new buttons and texts

[Test](https://mf-geoadmin3.dev.bgdi.ch/teo_25d/mobile.html?lang=fr&topic=ech&bgLayer=ch.swisstopo.pixelkarte-farbe&layers=ch.swisstopo.zeitreihen,ch.bfs.gebaeude_wohnungs_register,ch.bafu.wrz-wildruhezonen_portal,ch.swisstopo.swisstlm3d-wanderwege,ch.swisstopo.swissimage-product&layers_visibility=false,false,false,false,true&layers_timestamp=18641231,,,,&lon=8.51818&lat=47.40393&elevation=6608&heading=360.000&pitch=-40.020)

Missing:

 - [ ] Translate swissimage warning message
 - [ ] Help for 3d

### CODE REVIEW

**Comment by davidoesch** [comment]
change of plans
the warntext has to be active for *ALL* draped layer in ALPHA, he is not visible if only bg is active
(pls don't argue with me, it is how it is)

File: src/components/backgroundselector/style/backgroundselector.less [hunk 1]
```diff
@@ -1,256 +1,263 @@
...
-  .ga-disable3d {
-    cursor: default;
-    background-color: gray;
-    color: gray;
-    &:hover {
+
+  [ga-background-selector] {
+
+    .ga-disable3d {
+      cursor: default;
```
Comment (loicgasser) [inline src/components/backgroundselector/style/backgroundselector.less]: cursor: not-allowed ?

Comment (cedricmoullet) [inline src/components/backgroundselector/style/backgroundselector.less]: 1. The question marks are usually put on the right of the element. I would
also put it on the right of the 3D control.
2. warning text: fine for me.

2015-12-03 18:18 GMT+01:00 Gasser Loïc <notifications@github.com>:

> In src/components/backgroundselector/style/backgroundselector.less
> <https://github.com/geoadmin/mf-geoadmin3/pull/2958#discussion_r46581422>:
>
> >      }
> >    }
> >  }
> >
> >  .ga-3d-active {
> > -  .ga-disable3d {
> > -    cursor: default;
> > -    background-color: gray;
> > -    color: gray;
> > -    &:hover {
> > +
> > +  [ga-background-selector] {
> > +
> > +    .ga-disable3d {
> > +      cursor: default;
>
> cursor: not-allowed ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geoadmin/mf-geoadmin3/pull/2958/files#r46581422>.
>



-- 
Twitter: http://twitter.com/cedricmoullet
Linked In: http://www.linkedin.com/in/cedricmoullet

<http://twitter.com/cedricmoullet>

Comment (davidoesch) [inline src/components/backgroundselector/style/backgroundselector.less]: +1 for the question mark at the right side

**Comment by oterral** [comment]
Fine for me it's a lot easier to implement, if it's not a layer specific message

File: src/components/backgroundselector/style/backgroundselector.less (hunk 1 above)
Comment (oterral) [inline src/components/backgroundselector/style/backgroundselector.less]: yes @loicgasser cursor: not-alowed is more indicated.
yes @cedricmoullet @davidoesch I put it on the right.

**Comment by oterral** [comment]
PR updated

**Comment by gjn** [comment]
@davidoesch Translations for 3d_overlay_warning?

**Comment by gjn** [comment]
@pfanguin What's state of the help?

**Comment by davidoesch** [comment]
Translation next week

**Comment by pfanguin** [comment]
I am working on it, I think it's could be ready at the beginning of the next week

File: src/components/attribution/AttributionDirective.js [hunk 2]
```diff
@@ -98,4 +98,58 @@ goog.require('ga_map_service');
...
+            });
+            dereg = [];
+            element.hide();
+          }
+        };
+
+        scope.$watch('::ol3d', function(val) {
+          if (val) {
+            // Listen when the app switch between 2d/3d
+            scope.$watch(function() {
```
Comment (gjn) [inline src/components/attribution/AttributionDirective.js]: Don't we add multiple watches when toggling mutliple times between 2d/3d?

File: src/index.mako.html [hunk 3]
```diff
@@ -717,9 +721,9 @@
         module.constant('gaGlobalOptions', {
           //dev3d to be removed once 3d goes live
 % if staging=='prod' :
-          dev3d: false,
+          dev3d: true,
```
Comment (gjn) [inline src/index.mako.html]: Please revert this for this branch so we can merge now. Final activation will be on 3dalpha branch.

**Comment by gjn** [comment]
@oterral Added a couple of comments.

I'd like to merge this for tomorrow.

File: src/components/attribution/AttributionDirective.js (hunk 2 above)
Comment (oterral) [inline src/components/attribution/AttributionDirective.js]: no `::ol3d` watcher above is called only one time

File: src/index.mako.html (hunk 3 above)
Comment (oterral) [inline src/index.mako.html]: yes

File: src/components/attribution/AttributionDirective.js (hunk 2 above)
Comment (gjn) [inline src/components/attribution/AttributionDirective.js]: aah. still not used to the `::` notation. sorry.

**Comment by oterral** [comment]
Done @gjn

**Comment by gjn** [comment]
Thanks alot.

**Pull request merged**